x-pack/plugins/observability_solution/logs_shared/common/log_views/resolved_log_view.ts

```diff
@@ -110,6 +119,52 @@ const resolveDataViewReference = async (
   };
 };
 
+const resolveKibanaAdvancedSettingReference = async (
+  logViewId: string,
+  logViewAttributes: LogViewAttributes,
+  dataViewsService: DataViewsContract,
+  logSourcesService: LogSourcesService
+): Promise<ResolvedLogView> => {
+  if (logViewAttributes.logIndices.type !== 'kibana_advanced_setting') {
+    throw new Error(
+      'This function can only resolve references to the Log Sources Kibana advanced setting'
+    );
+  }
+
+  const indices = (await logSourcesService.getLogSources())
+    .map((logSource) => logSource.indexPattern)
+    .join(',');
+
+  const dataViewReference = await dataViewsService
```

Comment:
> [!IMPORTANT]
> Just to double-check my understanding, this would create an ad-hoc data view and persist it in the in-memory cache only, not as a saved object right? Is this the expected behaviour? Does it work consistently across refreshes/navigation when using the embedded LogStream component in other solutions?

Reply:
Yep, this is just ad-hoc, there is no saved object. This is the same [as the logic for the indices option](https://github.com/elastic/kibana/blob/09f98ad0d13971c403ac29f9b13571f2ccc7f5a5/x-pack/plugins/observability_solution/logs_shared/common/log_views/resolved_log_view.ts#L55), essentially this just makes it easier to have a unified resolved log view interface alongside the data view option of a log view. 

This should work fine across the embeddable, the emeddable creates it's own log view client and follows the normal process for resolving a log view.